analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/Binary.cs

```diff
@@ -44,6 +46,52 @@ protected override ProgramState LearnBranchingConstraint(ProgramState state, IBi
         return state;
     }
 
+    protected override ProgramState PreProcess(ProgramState state, IBinaryOperationWrapper operation)
+    {
+        if (operation.LeftOperand is { } leftOperand
+            && operation.RightOperand is { } rightOperand
+            && state[leftOperand]?.Constraint<NumberConstraint>() is { } leftNumber
+            && state[rightOperand]?.Constraint<NumberConstraint>() is { } rightNumber
+            && Calculate(operation.OperatorKind, leftNumber, rightNumber) is { } newConstraint)
+        {
+            state = state.SetOperationConstraint(operation.WrappedOperation, newConstraint);
+        }
+        return state;
+    }
+
+    private static NumberConstraint Calculate(BinaryOperatorKind kind, NumberConstraint left, NumberConstraint right)
+    {
+        BigInteger? min, max;
+        switch (kind)
+        {
+            case BinaryOperatorKind.Add:
+                min = Calculation(kind, left.Min, right.Min);
+                max = Calculation(kind, left.Max, right.Max);
+                break;
+            case BinaryOperatorKind.Subtract:
+                min = Calculation(kind, left.Min, right.Max);
+                max = Calculation(kind, left.Max, right.Min);
+                break;
+            default:
+                min = null;
+                max = null;
+                break;
+        }
+        return min is not null || max is not null
+            ? NumberConstraint.From(min, max)
+            : null;
```

Comment:
I think this problem will re-apprear quite often. Let's update `From` to return `null` instead.

Reply:
Like that, we can remove `min` and `max` completely. 
Then, this can be a switch expression and arrow method.